R/calcFEbyEUEC.R

```diff
@@ -0,0 +1,208 @@
+#' it into different end uses.
+#'
+#' The disaggregation is performed such that the aggregated carrier-specific FE
+#' values of the IEA IO regions are met as a minimum requirement. In an ideal case,
+#' the enduse-specific FE shares are met as well.
+#'
+#' @note
+#' For now, existing disaggregated final energy data with respect to carriers and
+#' enduses combined is replaced in the final output. However, since the Odyssee
+#' data is largely underestimating the real IEA FE targets, these shall serve as
+#' lower boundaries for the disaggregation.
+#' Such feature has been implemented in \code{toolDisaggregate} but is not running
+#' smoothly yet.
+#'
+#' @returns data.frame with historic energy demands
+#'
+#' @author Hagen Tockhorn, Robin Hasse
+#'
+#' @importFrom dplyr mutate semi_join right_join left_join as_tibble filter
+#'   select %>% .data pull
+#' @importFrom madrat toolCountryFill calcOutput toolGetMapping
+#' @importFrom quitte as.quitte
+#' @importFrom magclass as.magpie
+#' @export
+
+calcFEbyEUEC <- function() {
+
+  # READ-IN DATA ---------------------------------------------------------------
+
+  # FE Data
+  ieaIO <- calcOutput("IOEdgeBuildings", subtype = "output_EDGE_buildings",
+                      aggregate = FALSE) %>%
+    as.quitte(na.rm = TRUE)
+
+
+  # FE EU Data
+  feOdyssee <- calcOutput("ShareOdyssee",
+                          subtype = "enduse_carrier",
+                          feOnly = TRUE,
+                          aggregate = FALSE) %>%
+    as.quitte()
+
+
+  # FE IEA EEI data
+  feIEAEEI <- calcOutput("IEA_EEI",
+                         subtype = "buildings",
+                         aggregate = FALSE) %>%
+    as.quitte()
+
+
+  # EU Shares
+  sharesEU <- calcOutput("Shares",
+                         subtype = "enduse_nonthermal",
+                         aggregate = TRUE,
+                         regionmapping = "regionmappingEUshares.csv") %>%
+    as.quitte()
+
+
+  # EU Shares mapping
+  regmapping <- toolGetMapping(name  = "regionmappingEUshares.csv",
+                               type  = "regional",
+                               where = "mredgebuildings") %>%
+    select("region", "regionAgg")
+
+
+
+  # PARAMETERS -----------------------------------------------------------------
+
+  # Enduse-Carrier combinations which will be systematically excluded
+  exclude <- toolGetMapping(name  = "excludeEnduseCarrier.csv",
+                            type  = "sectoral",
+                            where = "mredgebuildings")
+
+
+  # PROCESS DATA ---------------------------------------------------------------
+
+  sharesEU <- sharesEU %>%
+    select("region", "period", "enduse", "value")
+
+  ieaIO <- ieaIO %>%
+    rename(carrier = "variable") %>%
+    semi_join(sharesEU, by = c("period"))
+
+
+  # remove district space cooling from disaggregation
+  exclude <- exclude %>%
+    mutate(enduse = "space_cooling",
+           carrier = "heat")
```

Comment:
No, sorry, this was not intended. This enduse-carrier combination shall be added since results for the disaggregation looked bad and so far, EDGE-B assumed district cooling to be a future technology.